vermouth/graph_utils.py

```diff
@@ -216,7 +216,14 @@ def partition_graph(graph, partitions):
     for idx, jdx in graph.edges:
         if mapping[idx] != mapping[jdx]:
             new_idx, new_jdx = mapping[idx], mapping[jdx]
-            new_graph.add_edge(new_idx, new_jdx)
+            edge_attrs = graph.edges[(idx, jdx)]
+            if new_graph.has_edge(new_idx, new_jdx):
+                old_attrs = new_graph.edges[(new_idx, new_jdx)]
+                new_attrs = {key: old_attrs[key] for key in old_attrs.keys() & edge_attrs.keys()}
```

Comment:
This check is more relaxed than the one used for node attributes: there the values also have to be the same.